imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,187 @@
+            reading. For example, if the binary data contains a rotation flag,
+            the image is rotated by default and the rotation flag is excluded
+            from the metadata to avoid confusion.
+
+        Returns
+        -------
+        metadata : dict
+            A dictionary of format-specific metadata.
+
+        """
+
+        metadata = {}
+
+        ImageSize = self._image_file.sizes
+
+        # TO DO: Find out what information needs to be included below and refine accordingly.
+        metadata["black_level_per_channel"] = self._image_file.black_level_per_channel
+        metadata["camera_white_level_per_channel"] = self._image_file.camera_white_level_per_channel
+        metadata["color_desc"] = self._image_file.color_desc
+        #metadata["color_matrix"] = self._image_file.color_matrix
+        metadata["daylight_whitebalance"] = self._image_file.daylight_whitebalance
+        metadata["dtype"] = self._image_file.raw_image.dtype.type
+        metadata["flip"] = ImageSize.flip
+        metadata["num_colors"] = self._image_file.num_colors
+        #metadata["raw_colors"] = self._image_file.raw_colors
+        #metadata["raw_colors_visible"] = self._image_file.raw_colors_visible
+        #metadata["raw_image"] = self._image_file.raw_image
+        #metadata["raw_image_visible"] = self._image_file.raw_image_visible
+        #metadata["raw_pattern"] = self._image_file.raw_pattern
+        metadata["raw_type"] = self._image_file.raw_type.name
+        #metadata["rgb_xyz_matrix"] = self._image_file.rgb_xyz_matrix
+        #metadata["tone_curve"] = self._image_file.tone_curve
+        metadata["width"] = ImageSize.width
+        metadata["height"] = ImageSize.height
+        metadata["shape"] = (ImageSize.width, ImageSize.height)
+        metadata["raw_width"] = ImageSize.raw_width
+        metadata["raw_height"] = ImageSize.raw_height
+        metadata["raw_shape"] = self._image_file.raw_image.shape
+        metadata["iwidth"] = ImageSize.iwidth
+        metadata["iheight"] = ImageSize.iheight
+        metadata["pixel_aspect"] = ImageSize.pixel_aspect
+        metadata["white_level"] = self._image_file.white_level
+
+        #TO DO: Find out what information needs to be excluded below and refine accordingly.
+        if exclude_applied:
+            metadata.pop("flip", None)
+
+        return metadata
+    
+    def properties(self, index: int = None) -> ImageProperties:
+        """Standardized ndimage metadata
+
+        Returns
+        -------
+        properties : ImageProperties
+            A dataclass filled with standardized image metadata.
+
+        Notes
+        -----
+        This does not decode pixel data and is fast for large images.
+
+        """
+
+        ImageSize = self._image_file.sizes
+
+        width: int = ImageSize.width
+        height: int = ImageSize.height
+        shape: Tuple[int, ...] = (height, width)
+
+        dtype = self._image_file.raw_image.dtype.type
```

Comment:
Hm, I need to think about this one. You are forwarding the dtype of the _raw_ image here. However, all the methods (read/iter) provide the postprocessed image, which may have changed its type and thus no longer match what the type of the raw image. That said, I don't know if this can happen in rawpy, so I need to look into if this is the right thing to do, or not.

Reply:
One more thing, raw_image has raw_shape with raw_width and raw_height. The raw_height is slightly different than ImageSize.height of the raw image. In the documentation, there is ImageSize for width and height.  Do you think I should use raw_height instead of height from the ImageSize ? What will be the shape for both metadata and properties in that case ? Or should I keep them both as I have done now ?